src/lib.rs

```diff
@@ -330,6 +324,57 @@ impl BitRange {
     }
 }
 
+#[derive(Clone, Copy, Debug)]
+pub struct WriteConstraintRange {
+    pub min: u32,
+    pub max: u32,
+}
+
+impl WriteConstraintRange {
+    fn parse(tree: &Element) -> WriteConstraintRange {
+        WriteConstraintRange {
+            min: try!(try!(tree.get_child_text("minimum")).parse()),
+            max: try!(try!(tree.get_child_text("maximum")).parse()),
+        }
+    }
+}
+
+#[derive(Clone, Copy, Debug)]
+pub enum WriteConstraint {
+    Invalid,
```

Comment:
This is effectively an error. As we are lacking proper error handling, the parser should panic if this case is encountered.